src/redis.ts

```diff
@@ -0,0 +1,1256 @@
+    const command: string = "RPOP";
+    return this._commandParser(command, collection, key, count);
+  }
+
+  rpoplpush(source: string, destination: string, collection: string) {
+    const command: string = "RPOPLPUSH";
+    return this._commandParser(command, collection, source, destination);
+  }
+  // End of LIST commands
+
+  // Start of HASH commands
+  hset(key: string, data: object, collection: string) {
+    const dataArray: string[] = [];
+    for (const [key, value] of Object.entries(data)) {
+      dataArray.push(key);
+      dataArray.push(value);
+    }
+    const command: string = "HSET";
+    return this._commandParser(command, collection, key, ...dataArray);
+  }
+
+  hget(key: string, field: string, collection: string) {
+    const command: string = "HGET";
+    return this._commandParser(command, collection, key, field);
+  }
+
+  hdel(key: string, fields: string[], collection: string) {
+    const command: string = "HDEL";
+    return this._commandParser(command, collection, key, ...fields);
+  }
+
+  hexists(key: string, field: string, collection: string) {
+    const command: string = "HEXISTS";
+    return this._commandParser(command, collection, key, field);
+  }
+
+  hgetall(key: string, collection: string) {
+    const command: string = "HGETALL";
+    return this._commandParser(command, collection, key);
+  }
+
+  hincrby(key: string, field: string, increment: number, collection: string) {
+    const command: string = "HINCRBY";
+    return this._commandParser(command, collection, key, field, increment);
+  }
+
+  hincrbyfloat(
+    key: string,
+    field: string,
+    increment: number,
+    collection: string
+  ) {
+    const command: string = "HINCRBYFLOAT";
+    return this._commandParser(command, collection, key, field, increment);
+  }
+
+  hkeys(key: string, collection: string) {
+    const command: string = "HKEYS";
+    return this._commandParser(command, collection, key);
+  }
+
+  hlen(key: string, collection: string) {
+    const command: string = "HLEN";
+    return this._commandParser(command, collection, key);
+  }
+
+  hmget(key: string, fields: string[], collection: string) {
+    const command: string = "HMGET";
+    return this._commandParser(command, collection, key, ...fields);
+  }
+
+  hmset(key: string, data: object, collection: string) {
+    const dataArray: string[] = [];
+    for (const [key, value] of Object.entries(data)) {
+      dataArray.push(key);
+      dataArray.push(value);
+    }
+    const command: string = "HMSET";
+    return this._commandParser(command, collection, key, ...dataArray);
+  }
+
+  hscan(
+    key: string,
+    cursor: number,
+    collection: string,
+    pattern?: string | undefined,
+    count?: number | undefined
+  ) {
+    const command: string = "HSCAN";
+
+    let patternArray: any[] = [];
+    if (pattern !== undefined) {
+      patternArray.push("MATCH");
+      patternArray.push(pattern);
+    }
+
+    let countArray: any[] = [];
```

Comment:
same for ```countArray```. An array of numbers and strings.

Reply:
Found more instances in the code. Please review and update the types.

Reply:
Agree, will correct all instances, thanks!